components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -647,6 +652,104 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
+
+            if (local_config_GM_Griffies_compute_levels_every_timestep) then
+               ! config_GM_Griffies_compute_levels = 'every_timestep'
+
+               !$omp parallel
+               !$omp do schedule(runtime) private(k, cell1, cell2, sum_gradDensity, ltSum, countN2, &
+               !$omp             zEdge, ltAvg)
+               do iEdge = 1, nEdges
+                  sum_gradDensity = 0.0_RKIND
+                  ltSum = 0.0_RKIND
+                  countN2 = 0
+                  cell1 = cellsOnEdge(1, iEdge)
+                  cell2 = cellsOnEdge(2, iEdge)
+
+                  k = minLevelEdgeBot(iEdge)+1
+                  zEdge = -layerThickEdgeMean(k-1,iEdge)
+                  do while(zEdge > -config_GM_Griffies_min_depth .and. k < maxLevelEdgeTop(iEdge))
+                     k = k+1
+                     zEdge = zEdge - layerThickEdgeMean(k-1,iEdge)
+                  end do
+
+                  do while(zEdge > -config_GM_Griffies_max_depth .and. k < maxLevelEdgeTop(iEdge))
+                     ltAvg = 0.25_RKIND*(   layerThickness(k,cell1) + layerThickness(k-1,cell1) &
+                                          + layerThickness(k,cell2) + layerThickness(k-1,cell2))
+                     sum_gradDensity = sum_gradDensity + abs(gradDensityConstZTopOfEdge(k,iEdge))*ltAvg
+                     ltsum = ltsum + ltAvg
+                     countN2 = countN2 + 1
+
+                     k = k + 1
+                     zEdge = zEdge - layerThickEdgeMean(k-1,iEdge)
+                  end do
+
+                  if (countN2 > 0) then
+                     sum_gradDensity = sum_gradDensity / ltsum
+
+                     ! This is Griffies et al. (2005) eqn 15:
+                     ! A_{gm} = alpha |grad_z rho| L^2 g / ( rho_0 N_0)
+                     gmBolusKappa(iEdge) = coef * sum_gradDensity
+
+                     gmBolusKappa(iEdge) = max(config_GM_spatially_variable_min_kappa, min(gmBolusKappa(iEdge),  &
+                                               config_GM_spatially_variable_max_kappa))
+                 else !for really shallow columns use min bolus kappa
+                     gmBolusKappa(iEdge) = config_GM_spatially_variable_min_kappa
+                 end if
+               end do
+               !$omp end do
+               !$omp end parallel
+
+            else
+               ! config_GM_Griffies_compute_levels = 'init'
+
+               !$omp parallel
+               !$omp do schedule(runtime) private(k, cell1, cell2, sum_gradDensity, ltSum, countN2, &
+               !$omp             zEdge, ltAvg)
+               do iEdge = 1, nEdges
+                  sum_gradDensity = 0.0_RKIND
+                  ltSum = 0.0_RKIND
+                  countN2 = 0
+                  cell1 = cellsOnEdge(1, iEdge)
+                  cell2 = cellsOnEdge(2, iEdge)
+
+                  do k = max(Griffies_min_level,minLevelEdgeBot(iEdge)),min(Griffies_max_level,maxLevelEdgeTop(iEdge))
```

Comment:
@vanroekel That flag results in a simple loop here. I have your original code above. Let me know what you think of that.